### PR TITLE
[Feature](array-functions)improve array functions for array_last_index

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
Now we just support array_first_index for lambda input , but no array_last_index 
Issue Number: close #xxx

<--Describe your changes.-->

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/array/function_array_lambda_index.cpp [hunk 1]
```diff
@@ -45,11 +45,12 @@ class FunctionContext;
 namespace doris::vectorized {
 
 // array_first_index([0, 1, 0]) -> [2]
```
Comment (eldenmoon) [inline be/src/vec/functions/array/function_array_lambda_index.cpp:47]: add `array_last_index` annotation like `array_first_index`

File: be/src/vec/functions/array/function_array_lambda_index.cpp [hunk 2]
```diff
@@ -45,11 +45,12 @@ class FunctionContext;
...
-class FunctionArrayFirstIndex : public IFunction {
+template <bool first>
+class FunctionArrayLambdaIndex : public IFunction {
 public:
```
Comment (eldenmoon) [inline be/src/vec/functions/array/function_array_lambda_index.cpp:50]: if this class is just for `array_first_index` `array_last_index`, so i think this class could be renamed to `FunctionArrayFirstOrLastIndex` is better?  Or we need some enum

File: be/src/vec/functions/array/function_array_lambda_index.cpp [hunk 3]
```diff
@@ -88,10 +89,21 @@ class FunctionArrayFirstIndex : public IFunction {
 
             // default index is 0 if such index is not found
             size_t first_index = 0;
```
Comment (eldenmoon) [inline be/src/vec/functions/array/function_array_lambda_index.cpp:91]: rename first_index -> dst_index

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
rum p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**